### PR TITLE
python: Remove more code for <3.6

### PR DESCRIPTION
**Pull request by QuLogic**
## What does this PR do?

Removes a bunch of code for supporting Python < 3.6, applies lib2to3 conversions so it doesn't need to be run on build.

SWIG files could be regenerated to drop a bunch more code, but I don't have the same version, and moreover, I don't think those should be in version control.

## What are related issues/pull requests?

RFC 77

## Tasklist

 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

**Review by rouault — CHANGES_REQUESTED** [review]
As this is a non-trivial change, please regenerate the SWIG generated files (although I agree this isn't a ideal practice) with SWIG 3.0.12. Building SWIG from source is quite straighforward

File: gdal/gcore/gdalpython.cpp [hunk 1]
```diff
@@ -51,8 +51,8 @@ static std::mutex gMutex;
 static bool gbHasInitializedPython = false;
 static PyThreadState* gphThreadState = nullptr;
 
-static PyGILState_STATE (*PyLibGILState_Ensure)(void) = nullptr;
-static void (*PyLibGILState_Release)(PyGILState_STATE) = nullptr;
+static PyGILState_STATE (*PyGILState_Ensure)(void) = nullptr;
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply gdal/gcore/gdalpython.cpp:54]: Is there a reason those aren't in the `GDALPy` namespace? That would prevent the symbol clash too, I think.

[the remark this answers is not in the document]
Comment (rouault) [reply gdal/gcore/gdalpython.cpp:54]: > Is there a reason those aren't in the `GDALPy` namespace?

that would be better indeed !

**Review by rouault — APPROVED** [review]

File: gdal/swig/python/setup.py [hunk 2]
```diff
@@ -142,34 +115,16 @@ def fetch_config(option, gdal_config='gdal-config'):
...
-            raise Warning(p[2].readline())
+        p = subprocess.Popen([command, args], stdout=subprocess.PIPE)
+    except OSError:
+        e = sys.exc_info()[1]
+        raise gdal_config_error(e)
+    r = p.stdout.readline().decode('ascii').strip()
+    p.stdout.close()
+    p.wait()
 
     return r
```
Comment (rcoup) [inline gdal/swig/python/setup.py:129]:
```suggestion
        p = subprocess.run([command, args], stdout=subprocess.PIPE, check=True, encoding='ascii')
        return p.stdout.strip()
    except OSError as e:
        raise gdal_config_error(str(e)) from e
```

File: gdal/swig/python/GNUmakefile [hunk 3]
```diff
@@ -28,46 +28,63 @@ clean:
 	-rm -f *.so ./osgeo/*.so
 	-rm -rf dist
 
-SWIGARGS += -threads
+SWIGARGS += -relativeimport -threads
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply gdal/swig/python/GNUmakefile:31]: That mostly seems to just add type hints, but they seem incomplete.

File: gdal/swig/python/osgeo/gdal_array.py [hunk 4]
```diff
@@ -95,7 +95,12 @@ class _object:
         pass
     _newclass = 0
 
-import osgeo.gdal
+from sys import version_info as _swig_python_version_info
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply gdal/swig/python/osgeo/gdal_array.py:98]: Unfortunately yes, at least with 3.0.12.

File: gdal/swig/python/setup.py (hunk 2 above)
Comment (QuLogic) [reply gdal/swig/python/setup.py:129]: There are a whole bunch of these; I'd rather clean them all up in one go.

**Pull request merged**